### ISSUE TITLE
v1.0.7 debug Seven Seas

### ISSUE THREAD

**Issue by Telerex** [nysthi/nysthi#253]
I have had to turn of multithreading in Seven Seas since the debug. I am getting nothing but audio crackle with it on. 
Audio example included 
[Seven Seas Multithreading, off then on.zip](https://github.com/nysthi/nysthi/files/3825012/Seven.Seas.Multithreading.off.then.on.zip)

**Comment by nysthi** [comment]
did you try to change block size ?

try with higher number

what's your computer and cpu ?

**Comment by Telerex** [comment]
The example I sent was done with a block size of 256 samples. A block size of 16 vastly improves things but still crackles.
Windows 10 I7 4.20GHz
As a reminder, I'm the guy that did the Seven Seas Prophet VS. That ran four Poly Seven Seas. This time I'm only running two ordinary Seven Seas. Things were fine before the latest update.
I will give the Prophet VS patch a try when I get a chance.

**Comment by nysthi** [comment]
understand, but I've changed just the timing for the "sleep" in the extra thread

generally the more (512 sample) the better (this is happening on my mac, I can use fluidly 256 and 512 blocks without clicks)

**Comment by nysthi** [comment]
so you are using SevenSeas (not the POLY7 seas)
(trying to understand where can be the problem)

**Comment by Telerex** [comment]
Yes, the original Seven Seas.
Had a good chance to mess about today and the same things happens with Poly Seven Seas. I even deleted the nysthi rack folder and let rack install it again just incase.
After a good deal of testing, it seams that the problem is worst when the processing block size is set to 256 samples. 512 samples improves it ever so slightly and with Poly Seven Seas 1024 or 2048 samples gets rid of the crackle completely.
While testing I also noticed with Seven Seas that when multithreading is turned off, the pulse width control has no effect, but it's fine with Poly Seven Seas.
I hope this helps.

**Comment by nysthi** [comment]
please open a separate issue for 
`While testing I also noticed with Seven Seas that when multithreading is turned off, the pulse width`

**Comment by nysthi** [comment]
I have done a new release 1.0.8
I've added a new contextual menu "Tight loop mode"

if ON tries to emulate the sleeping time of the thread as in the 1.0.5

let me know!

**Comment by nysthi** [comment]
I've added 1024 and 2048 also to 7seas

**Comment by Telerex** [comment]
The tight loop seems to make no difference, where as the extra block sizes correct the problem. The only down side to that is the fact that 256 is the default, when you add Seven Seas or initialize the module you're back to the crackle. Maybe a set as default is needed.
Are Seven Seas and the poly version based on exactly the same code or was the poly version completely reworked, because multithreading effects the CPU meter in rack differently. What I mean by that is that when multithreading is off Seven Seas gives a higher percentage than Poly Seven Seas, but the opposite happens when multithreading is on, Seven Seas goes down and Poly Seven Seas goes up. I've included pictures to show what I mean.
I hope that I'm not wasting your time with this bit, but my non-coding head is trying to help keep Seven Seas as great as it is. 

![Multithreading off](https://user-images.githubusercontent.com/55559146/68550887-17db1900-03ff-11ea-8657-49d29aa7891e.JPG)
![Multithreading on](https://user-images.githubusercontent.com/55559146/68550891-21fd1780-03ff-11ea-9e70-e867ff1bffc8.JPG)

**Comment by nysthi** [comment]
yes, because the P7Seas copies all the value to have a CORRECT rendering
(this part is missing in 7seas, and changes are probably are less fluid)
and there many differences internally too 

remember that increasing the sample block increases the delay of the signal

at 44100 a 1024 block is 23 msec of delay

and as for me 256 are ok on a very old mac, I'll leave 256 as default

thank for support

**Comment by Kautenja** [comment]
Hi! I'm also getting crackling / popping from 7seas to unfortunately. For me, the issue not related to multiprocessing. I'm able to resolve the issue by setting the engine sample rate to 44.1kHz. However, using any other sample rate causes the clicking. I believe a simple way to resolve this issue temporarily would be to detect sample rates lower/higher than 44.1kHz and [up/down sample the incoming inputs to 44.1kHz using the SampleRateConverter class](https://vcvrack.com/docs/structrack_1_1dsp_1_1SampleRateConverter).

**Comment by nysthi** [comment]
I use already a SR converter, it's the same in all my samplers
on my 2011 mac works flawlessly

**Comment by Kautenja** [comment]
Hmm I hadn't looked at the code so I was unaware, but that is odd. I'm getting the issue on MacOS 10.15.2 and Ubuntu 19.04. On Ubuntu the stuttering seems more random for some reason.
